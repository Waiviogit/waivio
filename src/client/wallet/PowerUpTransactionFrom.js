import React from 'react';
import PropTypes from 'prop-types';
import { FormattedRelative } from 'react-intl';
import BTooltip from '../components/BTooltip';
import { epochToUTC } from '../helpers/formatter';
import { getTransactionDescription } from './WalletHelper';

const PowerUpTransactionFrom = ({ timestamp, amount, from, to, transactionType }) => {
  const options = { from };
  const description = getTransactionDescription(transactionType, options);
  return (
    <div className="UserWalletTransactions__transaction">
      <div className="UserWalletTransactions__icon-container">
        <i className="iconfont icon-flashlight_fill UserWalletTransactions__icon" />
      </div>
      <div className="UserWalletTransactions__content">
        <div className="UserWalletTransactions__content-recipient">
          {to === from ? (
            <React.Fragment>
              {description.powerUpTransaction}
              <span className="UserWalletTransactions__payout">{amount}</span>
            </React.Fragment>
          ) : (
            <React.Fragment>
              {description.powerUpTransactionFrom}
              <span className="UserWalletTransactions__received">
                {'+ '}
                {amount}
              </span>
            </React.Fragment>
          )}
        </div>
        <span className="UserWalletTransactions__timestamp">
          <BTooltip
            title={
              <span>
                <FormattedRelative value={epochToUTC(timestamp)} />
              </span>
            }
          >
            <span>
              <FormattedRelative value={epochToUTC(timestamp)} />
            </span>
          </BTooltip>
        </span>
      </div>
    </div>
  );
};

PowerUpTransactionFrom.propTypes = {
  timestamp: PropTypes.number.isRequired,
  amount: PropTypes.element.isRequired,
  from: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  transactionType: PropTypes.string.isRequired,
};

export default PowerUpTransactionFrom;
