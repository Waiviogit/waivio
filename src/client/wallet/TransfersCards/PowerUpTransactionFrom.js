import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FormattedNumber } from 'react-intl';
import { isInteger } from 'lodash';
import { getTransactionDescription } from '../WalletHelper';
import CardsTimeStamp from './CardsTimeStamp';

const PowerUpTransactionFrom = ({
  timestamp,
  amount,
  from,
  to,
  transactionType,
  symbol,
  currentUser,
}) => {
  const options = { from, to };
  const description = getTransactionDescription(transactionType, options);
  const isSent = currentUser === to;
  const amountClassList = classNames({
    UserWalletTransactions__received: isSent,
    UserWalletTransactions__sent: !isSent,
  });
  const minimumFractionDigits = isInteger(Number(amount)) ? 0 : 3;

  return (
    <div className="UserWalletTransactions__transaction">
      <div className="UserWalletTransactions__icon-container">
        <i className="iconfont icon-flashlight_fill UserWalletTransactions__icon" />
      </div>
      <div className="UserWalletTransactions__content">
        <div className="UserWalletTransactions__content-recipient">
          {to === from ? (
            <React.Fragment>
              <div>{description.powerUpTransaction}</div>
              <span className="UserWalletTransactions__payout-black">
                <FormattedNumber
                  value={amount}
                  locale={'en-IN'}
                  minimumFractionDigits={minimumFractionDigits}
                  maximumFractionDigits={3}
                />{' '}
                {symbol === 'WAIV' ? 'WP' : symbol}
              </span>
            </React.Fragment>
          ) : (
            <React.Fragment>
              {!isSent ? description.powerUpTransactionTo : description.powerUpTransactionFrom}
              <span className={amountClassList}>
                {!isSent ? '- ' : '+ '}
                {amount} {isSent ? 'WP' : symbol}
              </span>
            </React.Fragment>
          )}
        </div>
        <CardsTimeStamp timestamp={timestamp} />
      </div>
    </div>
  );
};

PowerUpTransactionFrom.propTypes = {
  currentUser: PropTypes.string.isRequired,
  timestamp: PropTypes.number.isRequired,
  amount: PropTypes.element.isRequired,
  from: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  transactionType: PropTypes.string.isRequired,
};

export default PowerUpTransactionFrom;
