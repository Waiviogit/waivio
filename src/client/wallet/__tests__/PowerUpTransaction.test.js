import React from 'react';
import { shallow } from 'enzyme';
import PowerUpTransactionFrom from '../PowerUpTransactionFrom';

describe('(Component) PowerUpTransactionFrom', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        timestamp: '0',
        amount: <span>{'0 HIVE'}</span>,
        to: 'hellosteem',
        from: 'hellosteem',
        transactionType: 'transfer_to_vesting',
      };
      const wrapper = shallow(<PowerUpTransactionFrom {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
