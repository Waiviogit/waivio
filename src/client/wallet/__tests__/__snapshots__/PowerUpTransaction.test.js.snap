// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) PowerUpTransactionFrom with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PowerUpTransactionFrom
    amount={
      <span>
        0 HIVE
      </span>
    }
    from="hellosteem"
    timestamp="0"
    to="hellosteem"
    transactionType="transfer_to_vesting"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserWalletTransactions__icon-container"
        >
          <i
            className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
          />
        </div>,
        <div
          className="UserWalletTransactions__content"
        >
          <div
            className="UserWalletTransactions__content-recipient"
          >
            <UNDEFINED>
              <FormattedMessage
                defaultMessage="Powered up "
                id="powered_up"
                values={Object {}}
              />
              <span
                className="UserWalletTransactions__payout"
              >
                <span>
                  0 HIVE
                </span>
              </span>
            </UNDEFINED>
          </div>
          <span
            className="UserWalletTransactions__timestamp"
          >
            <BTooltip
              title={
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>
              }
            >
              <span>
                <FormattedRelative
                  updateInterval={10000}
                  value={0}
                />
              </span>
            </BTooltip>
          </span>
        </div>,
      ],
      "className": "UserWalletTransactions__transaction",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
          />,
          "className": "UserWalletTransactions__icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-flashlight_fill UserWalletTransactions__icon",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <UNDEFINED>
                <FormattedMessage
                  defaultMessage="Powered up "
                  id="powered_up"
                  values={Object {}}
                />
                <span
                  className="UserWalletTransactions__payout"
                >
                  <span>
                    0 HIVE
                  </span>
                </span>
              </UNDEFINED>
            </div>,
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>
              </BTooltip>
            </span>,
          ],
          "className": "UserWalletTransactions__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <UNDEFINED>
                <FormattedMessage
                  defaultMessage="Powered up "
                  id="powered_up"
                  values={Object {}}
                />
                <span
                  className="UserWalletTransactions__payout"
                >
                  <span>
                    0 HIVE
                  </span>
                </span>
              </UNDEFINED>,
              "className": "UserWalletTransactions__content-recipient",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />,
                  <span
                    className="UserWalletTransactions__payout"
                  >
                    <span>
                      0 HIVE
                    </span>
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "defaultMessage": "Powered up ",
                    "id": "powered_up",
                    "values": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                      0 HIVE
                    </span>,
                    "className": "UserWalletTransactions__payout",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "0 HIVE",
                    },
                    "ref": null,
                    "rendered": "0 HIVE",
                    "type": "span",
                  },
                  "type": "span",
                },
              ],
              "type": Symbol(react.fragment),
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BTooltip
                title={
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>
              </BTooltip>,
              "className": "UserWalletTransactions__timestamp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>,
                "title": <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "updateInterval": 10000,
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserWalletTransactions__icon-container"
          >
            <i
              className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
            />
          </div>,
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <UNDEFINED>
                <FormattedMessage
                  defaultMessage="Powered up "
                  id="powered_up"
                  values={Object {}}
                />
                <span
                  className="UserWalletTransactions__payout"
                >
                  <span>
                    0 HIVE
                  </span>
                </span>
              </UNDEFINED>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value={0}
                  />
                </span>
              </BTooltip>
            </span>
          </div>,
        ],
        "className": "UserWalletTransactions__transaction",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
            />,
            "className": "UserWalletTransactions__icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-flashlight_fill UserWalletTransactions__icon",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="UserWalletTransactions__content-recipient"
              >
                <UNDEFINED>
                  <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />
                  <span
                    className="UserWalletTransactions__payout"
                  >
                    <span>
                      0 HIVE
                    </span>
                  </span>
                </UNDEFINED>
              </div>,
              <span
                className="UserWalletTransactions__timestamp"
              >
                <BTooltip
                  title={
                    <span>
                      <FormattedRelative
                        updateInterval={10000}
                        value={0}
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>
                </BTooltip>
              </span>,
            ],
            "className": "UserWalletTransactions__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <UNDEFINED>
                  <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />
                  <span
                    className="UserWalletTransactions__payout"
                  >
                    <span>
                      0 HIVE
                    </span>
                  </span>
                </UNDEFINED>,
                "className": "UserWalletTransactions__content-recipient",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormattedMessage
                      defaultMessage="Powered up "
                      id="powered_up"
                      values={Object {}}
                    />,
                    <span
                      className="UserWalletTransactions__payout"
                    >
                      <span>
                        0 HIVE
                      </span>
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultMessage": "Powered up ",
                      "id": "powered_up",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <span>
                        0 HIVE
                      </span>,
                      "className": "UserWalletTransactions__payout",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "0 HIVE",
                      },
                      "ref": null,
                      "rendered": "0 HIVE",
                      "type": "span",
                    },
                    "type": "span",
                  },
                ],
                "type": Symbol(react.fragment),
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <BTooltip
                  title={
                    <span>
                      <FormattedRelative
                        updateInterval={10000}
                        value={0}
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>
                </BTooltip>,
                "className": "UserWalletTransactions__timestamp",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>,
                  "title": <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedRelative
                      updateInterval={10000}
                      value={0}
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "updateInterval": 10000,
                      "value": 0,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
